docs/source/languages/orquesta.rst

```diff
@@ -195,19 +194,80 @@ be run two different times, resulting in this execution graph:
+                 +-- parallel_task_1 -------+
+                 |                          |
+                 |                          +-- barrier_task (1) ---------------------+
+                 |                                                                    |
+                 +-- parallel_task_2 --+                                              |
+                 |                     |                                              |
+                 |                     +-- intermediate_task --+                      |
+                 |                                             |                      |
+                 |                                             +-- barrier_task (2) --+
+                 |                                                                    |
+                 +-- parallel_task_3 -------------------------------------------------+
+                                                                                      |
+                                                                                      +-- [finish]
+
+An alternative use case of join is to specify an integer value such as ``join: <integer>``
+instead of ``join: all``. In this use case, the join is satisified when the number of task
+transitioned into the join is greater than or equal to the value specified. Take the following
+workflow definition below, which is a revised version of the workflow from previous example.
+There are three tasks that run in parallel and will join at the barrier task. The join has a
+value of 2 which means the join will be satisfied when two out of the three parallel tasks
+complete and transition into the join task. The join will immediately run when the criteria
+is satisfied.
+
+.. code-block:: yaml
+
+    version: 1.0
+
+    tasks:
+      setup_task:
+        next:
+          - do:
+              - parallel_task_1
+              - parallel_task_2
+              - parallel_task_3
+
+      parallel_task_1:
+        action: core.noop
+        next:
+          - when: <% succeeded() %>
+            do: barrier_task
+
+      parallel_task_2:
+        action: core.noop
+        next:
+          - when: <% succeeded() %>
+            do: barrier_task
+
+      parallel_task_3:
+        action: core.noop
+        next:
+          - when: <% succeeded() %>
+            do: barrier_task
+
+      barrier_task:
+        join: 2
+        action: core.noop
+
+The following is the corresponding workflow execution graph.
+
+.. code-block:: none
+
+    =---- time (not to scale) ---->
+
+    setup_task --+
+                 |
+                 +-- parallel_task_1 --+
+                 |                     |
+                 +-- parallel_task_2 --+
+                 |                     |
+                 +-- parallel_task_3 --+
+                                       |
+                                       +-- barrier_task (only requires 2 of 3 tasks) --+
```

Comment:
I disagree. This is not accurate description of the workflow.  This shows that the parallel_task_1 is on its own and does not join with the other parallel tasks at the barrier_task. But in fact, the join is satisfied as long as 2 out of 3 parallel tasks complete regardless of which one.

Reply:
What about something like this?

```
.. code-block:: none

    =---- time (not to scale) ---->

    setup_task --+
                 |
                 +-- parallel_task_1 -------------------+
                 |                                      . (joins to barrier_task)
                 +-- parallel_task_2 --+                .
                 |                     |                .
                 +-- parallel_task_3 --+                .
                                       |                .
                                       +-- barrier_task (only requires 2 of 3 tasks) --+
                                                                                       |
                                                                                       +-- [finish]
```

I think we should make it visually apparent that `parallel_task_1` is started with the other parallel tasks, but since it isn't finished before the other two parallel tasks finish, it doesn't join to the `barrier_task` with the other parallel tasks before that gets scheduled.

I use the dots to indicate that it joins `barrier_task` but isn't required. How does that look to you?

Reply:
We talked on Slack and compromised on using `.` instead of `|` to indicate the transition to `barrier_task` to differentiate this `join: 2` case from the `join: all` case.

Reply:
Ok I changed to all asterisk for all lines from parallel tasks to the barrier task.  Let's hope balance between accuracy and descriptive picture is good enough.